Build fails with Gradle 7.6

### Describe the bug

A build with gradle fails with the followin error message:
```
$ gradle :zap:distLinux
...
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildSrc:validatePlugins'.
> Plugin validation failed with 1 problem:
    - Warning: Type 'org.zaproxy.zap.tasks.GradleBuildWithGitRepos' property 'repositoriesDirectory' has @Input annotation used on property of type 'DirectoryProperty'.
      
      Reason: A property of type 'DirectoryProperty' annotated with @Input cannot determine how to interpret the file.
      
      Possible solution: Annotate with @InputDirectory for directories.
      
      Please refer to https://docs.gradle.org/7.6/userguide/validation_problems.html#incorrect_use_of_input_annotation for more details about this problem.
  See https://docs.gradle.org/7.6/userguide/more_about_tasks.html#sec:task_input_output_annotations for more information on how to annotate task properties.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 39s
18 actionable tasks: 8 executed, 1 from cache, 9 up-to-date
```

### Steps to reproduce the behavior

1. Build the project with `gradle :zap:distLinux`.
2. The build fails.

### Expected behavior

The build is successful.

### Software versions

My gradle version:
```
gradle --version

------------------------------------------------------------
Gradle 7.6
------------------------------------------------------------

Build time:   2022-11-25 13:35:10 UTC
Revision:     daece9dbc5b79370cc8e4fd6fe4b2cd400e150a8

Kotlin:       1.7.10
Groovy:       3.0.13
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          17.0.5 (Alpine 17.0.5+8-alpine-r2)
OS:           Linux 6.1.3-0-edge amd64
```

### Screenshots

_No response_

### Errors from the zap.log file

_No response_

### Additional context

_No response_

### Would you like to help fix this issue?

- [X] Yes

--- a/buildSrc/src/main/java/org/zaproxy/zap/tasks/GradleBuildWithGitRepos.java
+++ b/buildSrc/src/main/java/org/zaproxy/zap/tasks/GradleBuildWithGitRepos.java
@@ -48,6 +48,7 @@ import org.gradle.api.provider.ListProperty;
 import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.Console;
 import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputDirectory;
 import org.gradle.api.tasks.InputFile;
 import org.gradle.api.tasks.Nested;
 import org.gradle.api.tasks.TaskAction;
@@ -80,7 +81,7 @@ public class GradleBuildWithGitRepos extends DefaultTask {
         return repositoriesDataFile;
     }
 
-    @Input
+    @InputDirectory
     public DirectoryProperty getRepositoriesDirectory() {
         return repositoriesDirectory;
     }

